src/main/java/org/haffson/adventofcode/days/day01/Day01.java

```diff
@@ -1,32 +1,33 @@
 package org.haffson.adventofcode.days.day01;
 
+import org.haffson.adventofcode.utils.DataLoader;
 import org.haffson.adventofcode.ProblemStatusEnum;
 import org.haffson.adventofcode.days.Days;
+import org.haffson.adventofcode.utils.ProblemStatus;
 import org.springframework.stereotype.Component;
-import java.io.FileNotFoundException;
-import java.io.InputStream;
+
 import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * Implementation for <i>Day 1: Chronal Calibration</i>.
  */
 @Component
 public class Day01 implements Days {
 
-    /**
-     * The puzzle status {@code HashMap}
-     */
     private final Map<Integer, ProblemStatusEnum> problemStatus;
+    private final List<Integer> numbers;
 
-    // Read content of input file
-    public InputStream resource = getClass().getResourceAsStream("/data/day01/input_day01.txt");
-    private final Integer[] data = getRawData(resource);
+    public Day01(String filename) {
+        //get data
+        this.numbers = DataLoader.getDataDay01("/day01/" + filename, "\n");
```

Comment:
Gut, dass das Laden nun extern passiert. Hier wird DataLoader statisch genutzt, was den Nachteil hat, dass man ihn im Test immer mittestet und sich so von ihm und (wahrscheinlich) dem Dateisystem abhängig macht.

Reply:
Ja,
das stimmt, aber ich wollte dynamisch bleiben bzgl der Dateien, die ich laden möchte, denn ich habe pro Rätsel oftmals mehr als einen Dateiinput. Manchmal bis zu 4 (z.b. data part 1, data part 2, test data part 1, test data part 2). Ich habe lange überlegt, wie ich es am besten mache und habe mich für diese Form entschieden.

Reply:
Gibt es eine bessere Art mit verschiedenen Datensätzen umzugehen?

Reply:
Du könntest den `DataLoader` als Instanz an den Konstruktor übergeben und  die `getData`-Methode nicht statisch machen.
Dadurch kannst du den `DataLoader` im Test austauschen und durch eine Komponente ohne Dateisystemzugriff austauschen (z.B. ein Mock-Object).

Nächster Level: Anstatt einen `DataLoader` zu übergeben definierst du ein Interface, das `DataLoader` implementiert wird:

    public interface Loader {
        InputStream load(String path);
    }

    Day1(Loader dataLoader) {
    }

Wenn du das Interface als functional definierst, dann kannst du Lambdas anstelle von konkreten Instanzen benutzen:

    @FunctionalInterface
    public interface Loader {
        // ...
    }

Damit brauchst du in Tests nicht einmal Mock-Objekte:

    final var day1 = new Day((path) -> {
        // Hier Input-Stream erstellen, z.B. in-memory aus String.
        return input;
    });